dashboard/test/models/course_test.rb

```diff
@@ -262,7 +262,62 @@ class SelectCourseScriptTests < ActiveSupport::TestCase
     end
   end
 
-  class OtherVersionProgressTests < ActiveSupport::TestCase
+  class CanViewVersion < ActiveSupport::TestCase
+    setup do
+      @csp_2017 = create(:course, name: 'csp-2017', family_name: 'csp', version_year: '2017')
+      @csp1_2017 = create(:script, name: 'csp1-2017')
+      create :course_script, course: @csp_2017, script: @csp1_2017, position: 1
+      @csp_2018 = create(:course, name: 'csp-2018', family_name: 'csp', version_year: '2018')
+      @student = create :student
+    end
+
+    test 'teacher can always view version' do
+      assert @csp_2017.can_view_version?(create(:teacher))
+    end
+
+    test 'student can view version if it is the latest version in course family' do
+      assert @csp_2018.can_view_version?(@student)
+      refute @csp_2017.can_view_version?(@student)
+    end
+
+    test 'student can view version if it is assigned to them' do
+      create :follower, section: create(:section, course: @csp_2018), student_user: @student
+      create :follower, section: create(:section, course: @csp_2017), student_user: @student
+
+      assert @csp_2018.can_view_version?(@student)
+      assert @csp_2017.can_view_version?(@student)
+    end
+
+    test 'student can view version if they have progress in it' do
+      create :user_script, user: @student, script: @csp1_2017
+      assert @csp_2017.can_view_version?(@student)
+    end
+  end
+
+  class LatestVersionTests < ActiveSupport::TestCase
+    setup do
+      @csp_2017 = create(:course, name: 'csp-2017', family_name: 'csp', version_year: '2017')
+      @csp_2018 = create(:course, name: 'csp-2018', family_name: 'csp', version_year: '2018')
+      @student = create :student
+    end
+
+    test 'latest version returns nil if course family does not exist' do
+      assert_nil Course.latest_version('fake-family')
+    end
+
+    test 'latest version returns latest course version' do
+      latest_version = Course.latest_version('csp')
+      assert_equal @csp_2018, latest_version
+    end
+
```

Comment:
Will this have to be updated every year with a new version release?

Reply:
no, because the test database only knows that i've created 2017 and a 2018 versions, and that 2018 is the "latest" version created, it won't need to be updated